tests/sentry/flags/endpoints/test_hooks.py

```diff
@@ -0,0 +1,135 @@
+from datetime import datetime, timezone
+
+import pytest
+from django.urls import reverse
+
+from sentry.flags.endpoints.hooks import (
+    DeserializationError,
+    InvalidProvider,
+    handle_flag_pole_event,
+    handle_provider_event,
+)
+from sentry.flags.models import ACTION_MAP, CREATED_BY_TYPE_MAP, FlagAuditLogModel
+from sentry.testutils.cases import APITestCase
+from sentry.utils.security.orgauthtoken_token import hash_token
+
+
+def test_handle_provider_event():
+    result = handle_provider_event(
+        "flag-pole",
+        {
+            "data": [
+                {
+                    "action": "created",
+                    "flag": "test",
+                    "created_at": "2024-01-01T00:00:00",
+                    "created_by": "colton.allen@sentry.io",
+                    "tags": {"commit_sha": "123"},
+                }
+            ]
+        },
+        1,
+    )
+
+    assert result[0]["action"] == ACTION_MAP["created"]
+    assert result[0]["flag"] == "test"
+    assert result[0]["created_at"] == datetime(2024, 1, 1, 0, 0, 0, tzinfo=timezone.utc)
+    assert result[0]["created_by"] == "colton.allen@sentry.io"
+    assert result[0]["created_by_type"] == CREATED_BY_TYPE_MAP["email"]
+    assert result[0]["organization_id"] == 1
+    assert result[0]["tags"] == {"commit_sha": "123"}
+
+
+def test_handle_provider_event_invalid_provider():
+    with pytest.raises(InvalidProvider):
+        handle_provider_event("other", {}, 1)
+
+
+def test_handle_flag_pole_event():
+    result = handle_flag_pole_event(
+        {
+            "data": [
+                {
+                    "action": "created",
+                    "flag": "test",
+                    "created_at": "2024-01-01T00:00:00",
+                    "created_by": "colton.allen@sentry.io",
+                    "tags": {"commit_sha": "123"},
+                }
+            ]
+        },
+        1,
+    )
+
+    assert result[0]["action"] == ACTION_MAP["created"]
+    assert result[0]["flag"] == "test"
+    assert result[0]["created_at"] == datetime(2024, 1, 1, 0, 0, 0, tzinfo=timezone.utc)
+    assert result[0]["created_by"] == "colton.allen@sentry.io"
+    assert result[0]["created_by_type"] == CREATED_BY_TYPE_MAP["email"]
+    assert result[0]["organization_id"] == 1
+    assert result[0]["tags"] == {"commit_sha": "123"}
+
+
+def test_handle_flag_pole_event_bad_request():
+    try:
+        handle_flag_pole_event({"data": [{}]}, 1)
+    except DeserializationError as exc:
+        assert exc.errors["data"][0]["action"][0].code == "required"
+        assert exc.errors["data"][0]["flag"][0].code == "required"
+        assert exc.errors["data"][0]["created_at"][0].code == "required"
+        assert exc.errors["data"][0]["created_by"][0].code == "required"
+        assert exc.errors["data"][0]["tags"][0].code == "required"
```

Comment:
should tags be made optional?

Reply:
okay, just wondering- in case some providers don't have any tags by default, then they would have to send an empty dict every time. but not a huge deal if you want to keep it as required

Reply:
It wouldn't be required for those providers.  The schema is hard-coded per provider so different providers can (and will) submit a totally different field-set.

Reply:
Its coincidental hat the flag pole event matches the audit-log table (but not really a coincidence since we own both).